### ISSUE TITLE
dnsdist: No effect to disable the sending of ECS to the backend.

### ISSUE THREAD

**Issue by dmachard** [PowerDNS/pdns#13822]
- Program: dnsdist
 - Issue type: Bug report

### Short description

My purpose is to remove existing ECS option before to sending queries to the backend.
I'm trying to use the SetDisableECSAction to do that but without success.

### Environment
 - Software version: official docker image dnsdist 1.9.0

### Steps to reproduce

my basic config 

```
setLocal('0.0.0.0:53', { reusePort=true })
setACL({'0.0.0.0/0'})

newServer({address = "8.8.8.8"})
addAction(AllRule(), SetDisableECSAction())
addAction(AllRule(), PoolAction(""))
```

make a dig with /32 ip

dig @127.0.0.1 www.google.com +subnet=192.0.2.42/32
...
; EDE: 49152: (Provided ECS includes 32 bits, but no more than 24 are allowed. https://developers.google.com/speed/public-dns/docs/ecs)
...

### Expected behaviour

ECS option removed from queries as explained in the documentation

![image](https://github.com/PowerDNS/pdns/assets/5562930/c5a5fb26-cc0b-424a-bad1-d2472e77b4be)

### Other information

My workaround is to set the prefix to 0

```
setECSSourcePrefixV4(0)
setECSOverride(true)
newServer({address = "8.8.8.8", useClientSubnet=true})
addAction(AllRule(), PoolAction(""))

```

**Comment by rgacogne** [comment]
Right, it's at the very least a documentation issue: `SetDisableECSAction` prevents dnsdist from adding ECS information to the query before forwarding it to the backend, but it will not remove any existing ECS information already present in the initial query.

**Comment by dmachard** [comment]
I try to force non-disclosure IP source and `SetDisableECSAction` seemed to be the most appropriate function to do that according to my comprehension of the documentation.

Regarding your feedback, I can make a minor PR to add more additional information in the doc.

**Comment by rgacogne** [comment]
I'm afraid we don't have a way to remove existing EDNS options from a query. It might make sense to add such a feature in the future. In the meantime I think `SetEDNSOptionAction` might be another option, as the work-around you have found only works for `EDNS Client Subnet`.
